src/main.cpp

```diff
@@ -3350,6 +3461,53 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, CBlockIn
         }
     }
 
+    // Validation for witness commitments.
+    // * We compute the witness hash (which is the hash including witnesses) of all the block's transactions, except the
+    //   coinbase (where 0x0000....0000 is used instead).
+    // * The coinbase scriptWitness is a stack of a single 32-byte vector, containing a witness nonce (unconstrained).
+    // * We build a merkle tree with all those witness hashes as leaves (similar to the hashMerkleRoot in the block header).
+    // * The must be at least one output whose scriptPubKey is a single 36-byte push, the first 4 bytes of which are
```

Comment:
+    // * ~~The~~There must be at least one output whose scriptPubKey is a single 36-byte push, the first 4 bytes of which are